packages/node_modules/@webex/internal-plugin-devices/src/devices.js

```diff
@@ -336,8 +429,62 @@ const Devices = WebexPlugin.extend({
   },
 
   getWebSocketUrl() {}, // Retrieves the priority-mapped web socket url.
-  processRegistrationSuccess() {}, // Handles post-registration data mapping.
-  resetLogoutTimer() {}, // Resets the logout timer for ephemeral devices.
+
+  /**
+   * Process a successful device registration.
+   *
+   * @param {Object} response - response object from registration success.
+   * @returns {void}
+   */
+  processRegistrationSuccess(response) {
+    this.logger.info('devices: received registration payload');
+
+    // Assign the recieved DTO from **WDM** to this device.
+    this.set(response.body);
+
+    // Validate if device is ephemeral and setup refresh timer.
+    if (this.config.ephemeral) {
+      this.logger.info('devices: enqueuing device refresh');
+
+      const delay = (this.config.ephemeralDeviceTTL / 2 + 60) * 1000;
+
+      this.refreshTimer = safeSetTimeout(() => this.refresh(), delay);
+    }
+
+    // Emit the registration:success event.
+    this.trigger('registration:success', this);
```

Comment:
Do we have a constant for these events?

Reply:
There is not, I'll generate the constants as I believe they are much cleaner and allow for scalability.